weave

```diff
@@ -600,34 +600,67 @@ dns_args() {
     DNS_ARGS="$DNS_ARG $DNS_SEARCH_ARG"
 }
 
-# Perform operation $1 on local DNS database, for container ID $2
-# with an FQDN obtained from the container's configured hostname and
-# domain, at addresses $3, $4 ...
-tell_dns() {
-    METHOD="$1"
-    CONTAINER_ID="$2"
-    shift 2
-    CONTAINER_FQDN=$(docker inspect --format='{{.Config.Hostname}}.{{.Config.Domainname}}.' $CONTAINER_ID 2>/dev/null) || true
-    tell_dns_fqdn $METHOD $CONTAINER_ID $CONTAINER_FQDN $@
+# Execute $@ only if the weaveDNS container is running
+if_dns_running() {
+    if status=$(docker inspect --format='{{.State.Running}}' $DNS_CONTAINER_NAME 2>/dev/null) && [ "$status" = "true" ] ; then
+        "$@"
+    fi
 }
 
-# Perform operation $1 on local DNS database, for container ID $2
-# with FQDN $3, at addresses $4, $5 ... This function is only called
-# where we know $2 is a valid container name
-tell_dns_fqdn() {
-    METHOD="$1"
-    CONTAINER_ID="$2"
-    CONTAINER_FQDN="$3"
-    shift 3
-    if ! status=$(docker inspect --format='{{.State.Running}}' $DNS_CONTAINER_NAME 2>/dev/null) || [ "$status" != "true" ] ; then
-        # weavedns not running - silently return
-        return
+# Iff the container in $1 has an FQDN, invoke $2 as a command passing
+# the FQDN as the first argument and $3.. as additional arguments
+if_has_fqdn() {
+    CONTAINER="$1"
+    COMMAND="$2"
+    shift 2
+
+    CONTAINER_FQDN=$(docker inspect --format='{{.Config.Hostname}}.{{.Config.Domainname}}.' $CONTAINER 2>/dev/null) || true
+
+    if echo "$CONTAINER_FQDN" | grep -qv '\.\.$' ; then
+        $COMMAND "$CONTAINER" "$CONTAINER_FQDN" "$@"
     fi
-    # get the long form of the container ID
-    CONTAINER=$(docker inspect --format='{{.Id}}' $CONTAINER_ID 2>/dev/null)
-    MORE_ARGS="--data-urlencode fqdn=$CONTAINER_FQDN"
+}
+
+# Convert the container name or short ID in $2 to its full form, then
+# invoke $1 as a command passing the full form as the first argument and
+# $3.. as additional arguments
+with_full_container_id() {
+    CONTAINER="$1"
+    COMMAND="$2"
+    shift 2
+
+    $COMMAND $(docker inspect --format='{{.Id}}' $CONTAINER 2>/dev/null) "$@"
+}
```

Comment:
As discussed, tried alternative using shortcircuit evaluation in combination with `$(container_id ...)` but this is incompatible with failing gracefully when an unknown container is passed in. It also means all the statements gain an unsightly `|| true` at the end of the line to prevent failure of the overall command if weaveDNS isn't running. Consequently, am leaving as is an augmenting `with_full_container_id` to exit if the container doesn't exist.